### DISCUSSION TITLE
About view and edit icon hiding

### DISCUSSION THREAD

**Discussion by w1ndys** [squidfunk/mkdocs-material#6989] [Q&A] [answered]
Why did I set it in `index.md`
```
hide:
   - tags
   - navigation
   - footer
   - toc
   - edit
   - view
```
But the edit and view icons still exist

**Comment by w1ndys** [comment]
I think this might be needed
https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks
But I don't know how to edit it, can you teach me how to write it? I want to learn how to edit this extension

**Comment by w1ndys** [comment]
I want to find out how to edit HTML template files.
Is he here?
https://www.mkdocs.org/dev-guide/themes/

**Comment by alexvoss** [comment]
Hi, I assume you want to hide the [code action icons](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository#code-actions) on a specific page only? 

I don't think that is configurable via the page header. You could [override the actions partial](https://squidfunk.github.io/mkdocs-material/customization/#overriding-partials) and add a test to see if the page metadata has a hide entry for actions:

```
{% if page.edit_url %}
  {% if not 'actions' in page.meta.hide %}
  ...
  {% endif %}
{% endif %}
```

Don't forget the [register the overrides directory in your `mkdocs.yml`](https://squidfunk.github.io/mkdocs-material/customization/#setup-and-theme-structure).

**Reply by w1ndys** [reply]
You gave me a great idea. I tried to do it, but I found it didn't work. 
https://github.com/W1ndys/xkzb.qfnu.w1ndys.top/blob/main/overrides%2Fhide_code.html
https://github.com/W1ndys/xkzb.qfnu.w1ndys.top/blob/main/docs/index.md
Could you please tell me more about it?

In addition, I want to learn how to edit the override template file, but I haven't found any detailed development documents. Can you provide some learning materials? In other words, I want to learn how to write the files in override, because I don't know what variables to write in it to be effective.

**Reply by alexvoss** [reply, marked as answer]
You don't need to define a custom template because the code I posted contains a conditional that checks the page header for each page rendered. Copy the `actions.html` from your Material install (or [from here](https://raw.githubusercontent.com/squidfunk/mkdocs-material/master/src/templates/partials/actions.html) - assuming you use the latest version of Material) over to your partials directory. 

Then add the conditional into this. The first line in the code snippet I posted is already in the file, I posted that to show where the second line goes that tests the page header. Don't forget to close the conditional at the end with `{% endif %}`.

If you want to learn how to modify the templates and partials, the place to look is the [Jinja documentation](https://jinja.palletsprojects.com/en/3.1.x/). It also helps to have a good look at the templates that are part of Material.

**Reply by alexvoss** [reply]
Here's my example project zipped up: 
[actions.zip](https://github.com/squidfunk/mkdocs-material/files/14807788/actions.zip)

**Comment by w1ndys** [comment]
@alexvoss 
Thank you for your help. I have another question to ask.
That is, I want to make a front page similar to the official [material](https://github.com/squidfunk/mkdocs-material/discussions/6989#discussioncomment-8954921) document for my Mkdocs. Do I need to use a template extension?
How should I edit the related Markdown and HTML as well as css and how can I adapt to the whole framework (for example, the navigation on the page will not be affected after the page is designed)?

I found that there is not much explanation about how to edit syntax and define variables in the official document of material. maybe I didn't find it?

According to what you said above, should I learn from [jinja](https://jinja.palletsprojects.com/en/3.1.x/)? Is the grammar document I am looking for right here?

**Reply by alexvoss** [reply]
Have you looked at the discussion forum? There have been discussions of the front page before. https://github.com/squidfunk/mkdocs-material/discussions/4687 gives a concise answer and a pointer to more info on how to build your own custom homepage template.

It depends on the kind of customization you have in mind whether you can get away with just some CSS tweaking (good for hiding things), need to use some JavaScript or need to modify the Jinja templates. It does make sense to at least get a sense for how MkDocs works and a key component of this is Jinja.

**Reply by w1ndys** [reply]
This is great, thank you very much for your help, it is greatly appreciated
Learning how to configure material is really an interesting journey
